homeassistant/components/tcp/sensor.py

```diff
@@ -105,6 +121,11 @@ def update(self):
                 )
                 return
 
+            if self._ssl_context is not None:
```

Comment:
```suggestion
            if self._ssl_context:
```

Reply:
Why do you suggest that? Just for fewer chars in the line?
An if without a condition implies bool() applied to an object.
If I just want to check that it is not None, I should check that and avoid any overhead bool() might have.

Reply:
My intention was more about readability

Reply:
I understand.
But were you aware of what I explained?
If not, do you still think it's worth to have that slight readability improvement when it has a clear cost?

Reply:
\<SmartAssMode\>
Sometime `if val:` is slieghtly faster than `if val is not None:`, but mostly it is the same :nerd_face: 

```shell
$ python3 -m timeit -s"a='b'" "if a: pass" "else: pass"
20000000 loops, best of 5: 10.7 nsec per loop
```

```shell
$ python3 -m timeit -s"a='b'" "if a is not None: pass" "else: pass"
20000000 loops, best of 5: 10.7 nsec per loop
```

```shell
$ python3 -m timeit -s"a=None" "if a: pass" "else: pass"
20000000 loops, best of 5: 8.5 nsec per loop
```

```shell
$ python3 -m timeit -s"a='b'" "if a is not None: pass" "else: pass"
20000000 loops, best of 5: 10.7 nsec per loop
```
\</SmartAssMode\>

I agree that `if val:` is not the same as `if val is not None:` but in this case i would prefer the `if val:` because of readability

Reply:
Hehe now do it again with an instance of a class with sleep(10) in __bool__() definition or that affects the state of the instance.
I'm sure _is not None_ is quite readable as a generic approach when we don't want to care about the instance implementation.
Though, yes, I do agree most bool() implementations are quick.
Cheers on the feedback anyway, always good to share